eliot/tests/test_output.py

```diff
@@ -644,6 +644,29 @@ def test_filedestination_writes_json_bytes(self):
             [message1, message2])
 
 
+    def test_filedestination_flushes(self):
+        """
+        L{FileDestination} flushes after every write, to ensure logs get
+        written out even if the local buffer hasn't filed up.
+        """
+        class CustomException(Exception):
+            pass
+
+        def flush():
+            raise CustomException
+
+        message1 = {"x": 123}
+        bytes_f = BytesIO()
+        bytes_f.flush = flush
```

Comment:
:question: This isn't like you @itamarst ! :smile:  
Isn't there a way to test this using real files with fixed buffer sizes or with buffering disabled?

https://docs.python.org/2/library/io.html#io.open says:

> buffering is an optional integer used to set the buffering policy. Pass 0 to switch buffering off (only allowed in binary mode), 1 to select line buffering (only usable in text mode), and an integer > 1 to indicate the size of a fixed-size chunk buffer.

Perhaps if you supply a file with a known buffer size and log a message smaller than that size, you can assert that the content has nevertheless been flushed.